broker-daemon/models/block-order.js

```diff
@@ -476,6 +438,67 @@ class BlockOrder {
 
     return BlockOrder.fromStorage(blockOrderId, value)
   }
+
+  /**
+   * @param {Array<Order>} orders
+   * @returns {Object} res
+   * @returns {Big} res.inbound
+   * @returns {Big} res.outbound
+   */
+  static async activeAmountsForOrders (orders) {
+    const {
+      CREATED,
+      PLACED,
+      EXECUTING
+    } = OrderStateMachine.STATES
+    const response = {
+      inbound: Big(0),
+      outbound: Big(0)
+    }
+    return orders.reduce((acc, { order, state }) => {
+      // If the order is not in an active state , then we can simply skip it
+      if (![CREATED, PLACED, EXECUTING].includes(state)) {
+        return acc
+      }
+
+      if (state === OrderStateMachine.STATES.EXECUTING) {
+        acc.inbound = acc.inbound.plus(order.inboundFillAmount)
+        acc.outbound = acc.outbound.plus(order.outboundFillAmount)
+        return acc
+      } else {
+        acc.inbound = acc.inbound.plus(order.inboundAmount)
+        acc.outbound = acc.outbound.plus(order.outboundAmount)
+        return acc
+      }
+    }, response)
+  }
+
+  /**
+   * @param {Array<Fill>} fills
+   * @returns {Object} res
+   * @returns {Big} res.inbound
+   * @returns {Big} res.outbound
+   */
+  static async activeAmountsForFills (fills) {
+    const {
+      CREATED,
+      FILLED
+    } = FillStateMachine.STATES
+    const response = {
+      inbound: Big(0),
+      outbound: Big(0)
+    }
+    return fills.reduce((acc, { fill }) => {
+      // If the order is not in an active state , then we can simply skip it
+      if (![CREATED, FILLED].includes(fill.state)) {
```

Comment:
General question: is there a reason the `Fill` state is structured differently than the `Order` state? (`orders` have state on same level as `order` where `fills` have state as a property of `fill`)

Reply:
@bradleystachurski there should be no difference. I think the property exists on both, however this should be the same as the order example.

e.g. `{ fill, state }` in the reduce function and we check with `.includes(state)`